src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Tizen/MediaPlayerImpl.tizen.cs

```diff
@@ -0,0 +1,438 @@
+				}
+				await tcsForStreamInfo.Task;
+			}
+			await TaskPrepare;
+
+			var metadata = new Dictionary<string, string>
+			{
+				[nameof(StreamMetadataKey.Album)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Album),
+				[nameof(StreamMetadataKey.Artist)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Artist),
+				[nameof(StreamMetadataKey.Author)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Author),
+				[nameof(StreamMetadataKey.Genre)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Genre),
+				[nameof(StreamMetadataKey.Title)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Title),
+				[nameof(StreamMetadataKey.Year)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Year)
+			};
+			return metadata;
+		}
+
+		public async Task<MSize> GetVideoSize()
+		{
+			if (player.State == PlayerState.Idle)
+			{
+				if (tcsForStreamInfo == null || tcsForStreamInfo.Task.IsCompleted)
+				{
+					tcsForStreamInfo = new TaskCompletionSource<bool>();
+				}
+				await tcsForStreamInfo.Task;
+			}
+			await TaskPrepare;
+
+			var videoSize = player.StreamInfo.GetVideoProperties().Size;
+			return new MSize(videoSize.Width, videoSize.Height);
+		}
+
+		public View GetEmbeddingControlView(IMediaPlayer player)
+		{
+			return new EmbeddingControls
+			{
+				BindingContext = player
+			};
+		}
+
+		public void Dispose()
+		{
+			Dispose(true);
+			GC.SuppressFinalize(this);
+		}
+
+		protected virtual void Dispose(bool disposing)
+		{
+			if (disposed)
+				return;
+
+			if (disposing)
+			{
+				player.PlaybackCompleted -= OnPlaybackCompleted;
+				player.BufferingProgressChanged -= OnBufferingProgressChanged;
+				player.ErrorOccurred -= OnErrorOccurred;
+				player.Dispose();
+			}
+
+			disposed = true;
+		}
+
+		void ApplyDisplay()
+		{
+			if (VideoOutput == null)
+			{
+				player.Display = null;
+			}
+			else
+			{
+				if (player.Display != null)
+					return;
+
+				var renderer = Platform.GetRenderer(TargetView);
+				if (renderer is IMediaViewProvider provider && provider.GetMediaView() != null)
+				{
+					try
+					{
+						var display = new Display(provider.GetMediaView());
+						player.Display = display;
+						player.DisplaySettings.Mode = aspectMode.ToNative();
+					}
+					catch
+					{
+						Log.Error("Error on MediaView");
+					}
+				}
+			}
+		}
+
+		void ApplySource()
+		{
+			if (source == null)
+			{
+				return;
+			}
+
+			if (source is UriMediaSource uriSource)
+			{
+				var uri = uriSource.Uri;
+				if (uri != null)
+					player.SetSource(new MediaUriSource(uri.IsFile ? uri.LocalPath : uri.AbsoluteUri));
+			}
+			else if (source is FileMediaSource fileSource)
+			{
+				player.SetSource(new MediaUriSource(ResourcePath.GetPath(fileSource.File)));
+			}
+		}
+
+		void OnTargetViewPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == "Renderer")
+			{
+				if (Platform.GetRenderer(sender as BindableObject) != null && HasSource && AutoPlay)
```

Comment:
can we store the GetRenderer value in a local variable and use it here and in line 360?